types/reactstrap/lib/NavLink.d.ts

```diff
@@ -9,7 +9,7 @@ export interface NavLinkProps extends React.HTMLProps<HTMLAnchorElement> {
   cssModule?: CSSModule;
   onClick?: React.MouseEventHandler<any>;
   href?: string;
-  to?: string;
+  [others: string]: any;
```

Comment:
@deepu105 @timc13 

This has come up again. Thoughts?

Reply:
I am still against this and for generic components

Reply:
@FaithForHumans @timc13, unfortunately, I don't have enough time to keep up with this, so I give up. As maintainers, I assume you guys know what to do best here and hope that you will make the typings useful for everyone. For time being I have decided not use this typing

Reply:
So what am I supposed to do? All I know, that there is a "to" field in the NavLink component that wasn't found in the typings, and I needed this field in my project. I did the obvious and added the field and thought that hey, this could be useful to others as well, as seen in https://github.com/reactstrap/reactstrap/issues/298, so why not share it. That wasn't good enough and if the fix that was pointed to me wasn't good enough, then what is? It works with both solutions. Maybe sometimes purism is not the best way to handle things, but to deliver value and useful content to the user.

Reply:
If you look at the reactstrap source, [it doesn't care what `to` is](https://github.com/reactstrap/reactstrap/blob/master/src/NavLink.js#L6) it's not one of the props that a NavLink cares about. [Reactstrap passes along all props it doesn't care about](https://github.com/reactstrap/reactstrap/blob/master/src/NavLink.js#L50). Someone else might need `to` to be a boolean instead of a string if they're using something other than `react-router` for example.

Since reactstrap does pass along the other props, there needs to be a way to capture this fact as evidenced by this PR. There's two primary ways to do this: `[key: string]: any` or the generics method (what was in my previous comment above). That's what the discussion in #25329 was about, which is "better". Deepu105 pushed for the `[key: string]: any` method, and timc13 pushed for the generics method. You can read more detail in that PR.

I said "This has come up again" since it never really got resolved last time, and I didn't want to sneak through a PR without bringing it to their attention. Please don't misunderstand my intentions; I really do appreciate more people willing to pitch in, and I appreciate you bringing this PR forward as it got the ball rolling again.

If you have preference and more points for the `[key: string]: any` method over generics, please, field them here. Personally, I can see it going either way. That being said, I'll have time tomorrow to put together a PR to make all of the components generics that will fix this issue if we want to go down that path instead.

Reply:
As promised, #26094 contains the generics replacement.